libs/utils/env.py

```diff
@@ -439,6 +439,51 @@ def _init_target(self, force = False):
             self.__modules = ['bl', 'cpufreq']
             platform = Platform(model='pixel')
 
+        # Initialize gem5 platform
+        elif self.conf['board'].upper() == 'GEM5':
+            self.nrg_model = None
+            self.__modules=['cpufreq']
+
+            system_info = self.conf['gem5_settings']['system']
+            simulator_info = self.conf['gem5_settings']['simulator']
+            simulator_args = ''
+            platform_args = ''
+
+            # Get gem5 binary arguments
+            simulator_args = ' '.join(simulator_info.get('args', []) +
+                                      ['--listener-mode=on'])
+
+            # Get platform arguments
+            platform_args = ' '.join(system_info['platform'].get('args', []) +
+                                     ['--kernel {}'.format(system_info['kernel']),
+                                      '--dtb {}'.format(system_info['dtb']),
+                                      '--disk-image {}'.format(system_info['disk'])])
+
+            # Gather all arguments
+            args = ' '.join([simulator_args,
+                             system_info['platform']['description'],
+                             platform_args])
```

Comment:
This all "args" setup code maybe can be defined using a single args variable, which is what you really need for the `Gem5SimulationPlatform` despite being split in different fragments in the properties (all named `args`) in the configuration file:

```python
# gem5 arguments
args  = simulator.get('args', [])
args += ['--listener-mode=on']
# platform definition
args += system['platform']['description']
# platform arguments
args += system['platform'].get('args', [])
args += [
    '--kernel {}'.format(system['kernel']),
    '--dtb {}'.format(system['dsb']),
    '--disk-image {}'.format(system['disk-image'])
]

args = ' '.join(args)
   
```